great_expectations/validator/validation_graph.py

```diff
@@ -28,20 +40,190 @@ def id(self):
+                metrics.update(
+                    self._execution_engine.resolve_metrics(
+                        metrics_to_resolve=computable_metrics,
+                        metrics=metrics,
+                        runtime_configuration=runtime_configuration,
+                    )
+                )
+                progress_bar.update(len(computable_metrics))
+                progress_bar.refresh()
+            except ge_exceptions.MetricResolutionError as err:
+                if catch_exceptions:
+                    exception_traceback = traceback.format_exc()
+                    exception_message = str(err)
+                    exception_info = ExceptionInfo(
+                        exception_traceback=exception_traceback,
+                        exception_message=exception_message,
+                    )
+                    for failed_metric in err.failed_metrics:
+                        if failed_metric.id in failed_metric_info:
+                            failed_metric_info[failed_metric.id]["num_failures"] += 1  # type: ignore[operator]  # Incorrect flagging of 'Unsupported operand types for <= ("int" and "MetricConfiguration") and for >= ("Set[ExceptionInfo]" and "int")' in deep "Union" structure.
+                            failed_metric_info[failed_metric.id]["exception_info"].add(exception_info)  # type: ignore[union-attr]  # Incorrect flagging of 'Item "MetricConfiguration" of "Union[MetricConfiguration, Set[ExceptionInfo], int]" has no attribute "add" and Item "int" of "Union[MetricConfiguration, Set[ExceptionInfo], int]" has no attribute "add"' in deep "Union" structure.
+                        else:
+                            failed_metric_info[failed_metric.id] = {}
+                            failed_metric_info[failed_metric.id][
+                                "metric_configuration"
+                            ] = failed_metric
+                            failed_metric_info[failed_metric.id]["num_failures"] = 1
+                            failed_metric_info[failed_metric.id]["exception_info"] = {
+                                exception_info
+                            }
+                else:
+                    raise err
+            except Exception as e:
+                if catch_exceptions:
+                    logger.error(
+                        f"""Caught exception {str(e)} while trying to resolve a set of {len(ready_metrics)} metrics; aborting graph resolution."""
+                    )
+                    done = True
+                else:
+                    raise e
+
+            if (len(ready_metrics) + len(needed_metrics) == 0) or (
+                len(ready_metrics) == len(aborted_metrics_info)
+            ):
+                done = True
+
+        progress_bar.close()  # type: ignore[union-attr]  # Incorrect flagging of 'Item "None" of "Optional[Any]" has no attribute "close"' in external package.
```

Comment:
Again, you could use type-narrowing here to get rid of this warning.

Reply:
@Kilo59 Thanks -- as agreed, we can address in a follow-on PR.